ApiBundle/Transformer/AreaTransformer.php

```diff
@@ -96,82 +107,101 @@ public function transform($area, NodeInterface $node = null, $parentAreaId = nul
                 $otherNode->getId()
             ));
         }
-        $facade->boDirection = $area->getBoDirection();
-
-        $facade->uiModel = $this->getTransformer('ui_model')->transform(
-            array(
-                'label' => $area->getLabel(),
-                'class' => $area->getHtmlClass(),
-                'id' => $area->getAreaId()
-            )
-        );
-
-        $facade->addLink('_self_form', $this->generateRoute('open_orchestra_backoffice_area_form', array(
-            'nodeId' => $node->getId(),
-            'areaId' => $area->getAreaId(),
-        )));
 
-        if ($this->authorizationChecker->isGranted(TreeNodesPanelStrategy::ROLE_ACCESS_UPDATE_NODE, $node)) {
-            $facade->addLink('_self_block', $this->generateRoute('open_orchestra_api_area_update_block', array(
-                'nodeId' => $node->getId(),
-                'areaId' => $area->getAreaId()
+        if ($facade->editable) {
```

Comment:
@alavieille : this part of the code is extremely dense, could you use a protected method here to get the main code more readable ?